src/components/app/header/index.tsx

```diff
@@ -17,52 +17,319 @@ import SearchBar from './search-bar'
+      },
+      {
+        name: 'Vue.js',
+        href: '/q/vue',
+        image:
+          'https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/001/036/square_64/vue.png',
+      },
+      {
+        name: 'Next.js',
+        href: '/q/next',
+        image:
+          'https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/001/074/thumb/Group_1%281%29.png',
+      },
+      {
+        name: 'TypeScript',
+        href: '/q/typescript',
+        image:
+          'https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/000/377/square_64/typescriptlang.png',
+      },
+      {
+        name: 'JavaScript',
+        href: '/q/javascript',
+        image:
+          'https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/000/205/square_64/javascriptlang.png',
+      },
+      {
+        name: 'CSS',
+        href: '/q/css',
+        image:
+          'https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/000/175/square_64/csslang.png',
+      },
+    ],
+  },
+  {
+    name: 'Courses',
+    href: '/q?type=playlist',
+    image: (
+      <CodeIcon
+        className="text-gray-600 dark:text-gray-400"
+        strokeWidth={1.5}
+        height={20}
+        width={20}
+        aria-hidden="true"
+      />
+    ),
+  },
+  {
+    name: 'Tips',
+    href: '/tips',
+    image: (
+      <PlayIconOutline
+        className="text-gray-600 dark:text-gray-400"
+        strokeWidth={1.5}
+        height={20}
+        width={20}
+        aria-hidden="true"
+      />
+    ),
+  },
+  {
+    name: 'Guides',
+    href: '/guides',
+    image: (
+      <MapIconOutline
+        className=" text-gray-600 dark:text-gray-400"
+        strokeWidth={1.5}
+        height={20}
+        width={20}
+        aria-hidden="true"
+      />
+    ),
+  },
+  {
+    name: 'Articles',
+    href: '/blog',
+    image: (
+      <DocumentTextIconOutline
+        className="text-gray-600 dark:text-gray-400"
+        strokeWidth={1.5}
+        height={20}
+        width={20}
+        aria-hidden="true"
+      />
+    ),
+  },
+]
+
+const NavLink: React.FC<React.PropsWithChildren<NavLink>> = ({
+  name,
+  href,
+  items,
+  image,
+}) => {
+  const props = {
+    className:
+      'flex items-center h-full px-3 dark:hover:bg-white transition hover:bg-gray-50 gap-1 dark:hover:bg-opacity-5',
+    children: (
+      <>
+        {image}
+        {name}
+      </>
+    ),
+    onClick: () => {
+      track(`clicked ${name}`, {
+        location: 'header',
+      })
+    },
+  }
+
+  const LinkOrButton = () =>
+    items ? (
+      <NavDropdown
+        name={name}
+        items={items}
+        href={href}
+        className={props.className}
+      />
+    ) : href ? (
+      React.createElement(NextLink, {href, ...props})
+    ) : (
+      React.createElement('button', {
+        ...props,
+        onClick: () => {
+          props.onClick()
+          // do stuff
+        },
+      })
+    )
+
+  return <LinkOrButton />
+}
+
+const NavDropdown: React.FC<
```

Comment:
refactored this into generic dropdown component. I think we should be able to apply logic to transform it into an accordion component on mobile.